index.html

```diff
@@ -140,76 +189,254 @@ <h2>
+        "ltli/#locale">locale</a> 'en-NZ' but as '٧' in 'ar-EG'.
       </p>
     </section>
-    <section data-dfn-for="Badge">
+    <section>
       <h2>
-        <code>Badge</code> namespace
+        Extensions to the `Navigator` and `WorkerNavigator` interfaces
       </h2>
+      <p>
+        The {{Navigator}} and {{WorkerNavigator}} interfaces are extended with
+        methods for setting and clearing both the document and application
+        badge indicators from documents and service workers, respectively.
+      </p>
       <pre class="idl">
-        [Exposed=(Window)]
-        namespace Badge {
-          void set(optional [EnforceRange] unsigned long long contents);
-          void clear();
+        // Methods only exposed on documents.
+        [SecureContext]
+        partial interface Navigator {
+          Promise&lt;void&gt; setClientBadge(optional [EnforceRange] unsigned long long contents);
+          Promise&lt;void&gt; clearClientBadge();
+        };
+
+        // Methods exposed on both documents and service workers.
+        [SecureContext]
+        interface mixin NavigatorBadge {
+          Promise&lt;void&gt; setAppBadge(optional [EnforceRange] unsigned long long contents);
+          Promise&lt;void&gt; clearAppBadge();
         };
+
+        Navigator includes NavigatorBadge;
+        WorkerNavigator includes NavigatorBadge;
       </pre>
       <p>
-        The <dfn>Badge</dfn> namespace provides functions for setting and
-        clearing the application badge indicator.
+        User agents that never display document badges SHOULD NOT expose the
+        {{Navigator/setClientBadge()}} and {{Navigator/clearClientBadge()}}
+        methods. Similarly, user agents that never display application badges
+        SHOULD NOT expose the {{NavigatorBadge/setAppBadge()}} and
+        {{NavigatorBadge/clearAppBadge()}} methods.
+      </p>
+      <p class="note">
+        This is important as a feature-detection mechanism to allow sites to
+        determine whether setting a badge will have any effect, based on the
+        presence of the methods. In particular, sites can use the absence of
+        {{Navigator/setClientBadge()}} as a condition for displaying a fallback
+        badge in the page's [^title^] or <a data-cite=
+        "html#rel-icon"><code>icon</code></a>.
       </p>
+      <div class="issue" data-number="60"></div>
+      <div class="issue" data-number="61"></div>
+      <section>
+        <h3>
+          <dfn data-dfn-for="Navigator">setClientBadge()</dfn> method
+        </h3>
+        <p>
+          When the {{Navigator/setClientBadge()}} method is called on
+          |navigator:Navigator| with argument |contents:unsigned long long|:
+        </p>
+        <ol>
+          <li>Return [=a promise resolved with=] undefined, and [=in
+          parallel=]:
+            <ol>
+              <li>Let |document:Document| be |navigator|'s [=relevant settings
```

Comment:
HTML defines a "relevant settings object for a global object _O_", which we could perhaps use instead? it's currently not exported from HTML tho... @annevk or @domenic can probably confirm if that would be correct to use here?

Reply:
No, relevant settings object for a normal object seems right, since _navigator_ is presumably not the global object.

Reply:
@annevk Are you saying the current text (of this PR) is correct?

i.e. since this is a method on `navigator`, it should get `navigator`'s relevant settings object, not the global settings object.